src/file/properties.rs

```diff
@@ -0,0 +1,550 @@
+  pub fn set_statistics_enabled(mut self, value: bool) -> Self {
+    self.default_column_properties.set_statistics_enabled(value);
+    self
+  }
+
+  /// Sets max statistics size for any column.
+  /// Applicable only if statistics are enabled.
+  pub fn set_max_statistics_size(mut self, value: usize) -> Self {
+    self.default_column_properties.set_max_statistics_size(value);
+    self
+  }
+
+  // ----------------------------------------------------------------------
+  // Setters for a specific column
+
+  /// Helper method to get existing or new mutable reference of column properties.
+  #[inline]
+  fn get_mut_props(&mut self, col: ColumnPath) -> &mut ColumnProperties {
+    self.column_properties.entry(col).or_insert(ColumnProperties::new())
+  }
+
+  /// Sets encoding for a column.
+  /// Takes precedence over globally defined settings.
+  pub fn set_column_encoding(mut self, col: ColumnPath, value: Encoding) -> Self {
+    self.get_mut_props(col).set_encoding(value);
+    self
+  }
+
+  /// Sets compression codec for a column.
+  /// Takes precedence over globally defined settings.
+  pub fn set_column_compression(mut self, col: ColumnPath, value: Compression) -> Self {
+    self.get_mut_props(col).set_compression(value);
+    self
+  }
+
+  /// Sets flag to enable/disable dictionary encoding for a column.
+  /// Takes precedence over globally defined settings.
+  pub fn set_column_dictionary_enabled(mut self, col: ColumnPath, value: bool) -> Self {
+    self.get_mut_props(col).set_dictionary_enabled(value);
+    self
+  }
+
+  /// Sets flag to enable/disable statistics for a column.
+  /// Takes precedence over globally defined settings.
+  pub fn set_column_statistics_enabled(mut self, col: ColumnPath, value: bool) -> Self {
+    self.get_mut_props(col).set_statistics_enabled(value);
+    self
+  }
+
+  /// Sets max size for statistics for a column.
+  /// Takes precedence over globally defined settings.
+  pub fn set_column_max_statistics_size(mut self, col: ColumnPath, value: usize) -> Self {
+    self.get_mut_props(col).set_max_statistics_size(value);
+    self
+  }
+}
+
+/// Container for column properties that can be changed as part of writer.
+///
+/// If a field is `None`, it means that no specific value has been set for this column,
+/// so some default value must be used.
+#[derive(Debug, Clone, PartialEq)]
+struct ColumnProperties {
+  encoding: Option<Encoding>,
+  codec: Option<Compression>,
+  dictionary_enabled: Option<bool>,
+  statistics_enabled: Option<bool>,
+  max_statistics_size: Option<usize>
+}
+
+impl ColumnProperties {
+  /// Initialise column properties with default values.
+  fn new() -> Self {
+    Self {
+      encoding: None,
+      codec: None,
+      dictionary_enabled: None,
+      statistics_enabled: None,
+      max_statistics_size: None
+    }
+  }
+
+  /// Sets encoding for this column.
+  ///
+  /// This field is treated as fallback encoding, and should not be dictionary encoding.
```

Comment:
If the dictionary encoding is disabled for this column, then the parameter will be used as the primary encoding, correct?

Reply:
Yes, that is correct. If dictionary is enabled, this acts as fallback encoding. If dictionary is disabled, we will use this encoding as a primary. I will update the comment to make it clear.